packages/app/cypress/e2e/specs_page.cy.ts

```diff
@@ -0,0 +1,353 @@
+          cy.get('@CreateFromComponentDialog').within(() => {
+            cy.findByLabelText('file-name-input').as('FileNameInput')
+            .should('have.value', '')
+
+            cy.findByTestId('file-match-indicator').should('contain', '2 Matches')
+
+            cy.get('@FileNameInput')
+            .type('foobar')
+
+            cy.findByTestId('file-match-indicator').should('contain', 'No Matches')
+
+            cy.findByTestId('no-results-clear').click()
+
+            cy.findByTestId('file-match-indicator').should('contain', '2 Matches')
+
+            cy.get('@FileNameInput')
+            .type('App.jsx')
+
+            cy.findByTestId('file-match-indicator').should('contain', '1 of 2 Matches')
+          })
+        })
+
+        it('shows success modal with spec is created from component', () => {
+          cy.get('@CreateFromComponentDialog').within(() => {
+            cy.findAllByTestId('file-list-row').eq(0).as('NewSpecFile')
+
+            // TODO: assert visibility of secondary text on hover/focus when
+            // item is made keyboard accessible
+            // cy.get('@NewSpecFile).focus()
+            // cy.findByText('src/stories/Button.stories.jsx').should('be.visible')
+
+            cy.get('@NewSpecFile').click()
+          })
+
+          cy.findByRole('dialog', { name: defaultMessages.createSpec.successPage.header }).as('SuccessDialog').within(() => {
+            cy.validateExternalLink({ name: 'Need help?', href: 'https://on.cypress.io' })
+            cy.contains('src/App.cy.jsx').should('be.visible')
+            cy.findByRole('button', { name: 'Close' }).should('be.visible')
+            cy.findByRole('link', { name: 'Okay, run the spec' }).should('have.attr', 'href', '#/specs/runner?file=src/App.cy.jsx')
+            cy.findByRole('button', { name: 'Create another spec' }).click()
+          })
+
+          // 'Create a new spec' dialog presents with options when user indicates they want to create
+          // another spec.
+          cy.findByRole('dialog', { name: defaultMessages.createSpec.newSpecModalTitle }).within(() => {
+            cy.findAllByTestId('card').eq(0)
+            .should('have.attr', 'tabindex', '0')
+
+            // the storybook card remains disabled here
+            cy.findAllByTestId('card').eq(1)
+            .should('have.attr', 'tabindex', '-1')
+          })
+        })
+      })
+    })
+  })
+
+  describe('Code Generation', () => {
+    beforeEach(() => {
+      cy.scaffoldProject('react-code-gen')
+      cy.openProject('react-code-gen')
+      cy.startAppServer('component')
+      cy.visitApp()
+    })
+
+    const checkCodeGenCandidates = (specs: string[]) => {
+      cy.findByTestId('file-match-indicator').contains(`${specs.length} Match${specs.length > 1 ? 'es' : ''}`)
+      cy.findAllByTestId('file-list-row').should('have.length', specs.length)
+      .each((row, i) => cy.wrap(row).contains(specs[i]))
+    }
+
```

Comment:
It'd be worth beefing this up to test for:
- generating spec from component/story multiple times --  naming and code generation
- cannot generate spec from a spec
- generating a spec from a story when the `create from component` route is used